boards/DIM/Test/Inc/Test_SevenSegDisplays.h

```diff
@@ -13,7 +13,7 @@ DECLARE_FAKE_VOID_FUNC(set_middle_hex_digit, struct SevenSegHexDigit);
 DECLARE_FAKE_VOID_FUNC(set_left_hex_digit, struct SevenSegHexDigit);
 DECLARE_FAKE_VOID_FUNC(display_value_callback);
 
-class SevenSegDisplaysTest : public testing::Test
+class SevenSegDisplaysTest : public virtual testing::Test
```

Comment:
@garethellis0 Ran into the diamond problem where `class SevenSegDisplaysTest ` and `class RegenPaddleTest` inherit from the same base class `testing::Test`, then `class StateMachineTest` uses multiple inheritance. I did some quick research but honestly I don't know what's the right thing to do here: redesign the inheritance or just slap on a virtual inheritance?

Reply:
virtual inheritance seems fine to me here, this is one of the textbook cases for it.